crates/router/src/db/merchant_account.rs

```diff
@@ -524,7 +522,19 @@ impl MerchantAccountInterface for MockDb {
         _merchant_key_store: &domain::MerchantKeyStore,
     ) -> CustomResult<domain::MerchantAccount, errors::StorageError> {
         // [#TODO]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut accounts = self.merchant_accounts.lock().await;
+        let account = accounts
+            .iter_mut()
+            .find(|account| account.get_id() == _merchant_id)
+            .ok_or(errors::StorageError::ValueNotFound(format!(
+                "Merchant ID: {:?} not found",
+                _merchant_id
+            )))?;
+        account
+            .from_update(_merchant_account.into())
```

Comment:
Considering my comment on the `from_update()` method, can we avoid the macro implementation, and do something similar to the other implementations?

Here's the update implementation on profiles, for your reference:
https://github.com/juspay/hyperswitch/blob/4ef48c39b3ed7c1fcda9c850da766a0bdb701335/crates/router/src/db/business_profile.rs#L325-L364